apps/opencs/view/render/instancemode.cpp

```diff
@@ -728,36 +735,106 @@ void CSVRender::InstanceMode::dropInstance(DropMode dropMode, CSVRender::Object*
     objectNode->setNodeMask(oldMask);
 }
 
-void CSVRender::InstanceMode::dropSelectedInstances(bool active)
+float CSVRender::InstanceMode::getDropHeight(DropMode dropMode, CSVRender::Object* object)
 {
-    std::vector<osg::ref_ptr<TagBase> > selection = getWorldspaceWidget().getSelection (SceneUtil::Mask_EditorReference);
-    if (selection.empty()) return;
+    const osg::Vec3d& point = object->getPosition().asVec3();
+    osg::ref_ptr<osg::Group> objectNode = object->getRootNode();
+    osg::Node::NodeMask oldMask = objectNode->getNodeMask();
+    objectNode->setNodeMask(SceneUtil::Mask_Disabled);
```

Comment:
This is a bit of a problem. What happens when there is an exception between setting the new mask and restoring the old mask? The same problem couldd be present in the previous commit. Guess I missed it during the review.

This needs to be made exception-safe. A try-catch with a re-throw would be an option. A better option would be something like the CommandMacro.

Reply:
Ok, that's doable, but while testing this, I also found another issue: When dropping multiple objects, only one object at a time is set to Mask_Disabled, therefore dropped objects that are above each other tend to collide with each other. Correct behavior would be to set all dropped objects temporarily to Mask_Disabled, and everything in-between then done with try-catch.